graphql/graphql.go

```diff
@@ -254,14 +274,14 @@ func (t *Transaction) GasPrice(ctx context.Context) (hexutil.Big, error) {
 }
 
 func (t *Transaction) EffectiveGasPrice(ctx context.Context) (*hexutil.Big, error) {
-	tx, err := t.resolve(ctx)
-	if err != nil || tx == nil {
-		return nil, err
-	}
 	// Pending tx
 	if t.block == nil {
 		return nil, nil
 	}
```

Comment:
This still accesses `t.block`, which you set within the locked portion of `t.resolve()`. Race?

Reply:
I must admit I don't fully grokk the concurrency model of graphql

Reply:
Good catch. I also realized doing this pending check before resolve is wrong.